### PR TITLE
[AMORO-2410] Upgrade iceberg to 1.4.3

### LINKED ISSUE #2410: [Improvement]: Support iceberg 1.4.X
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

1.4.X starts using V2 format as default, update log [here](https://iceberg.apache.org/releases/#140-release)，It fixes many bugs and. By the way, the latest version is 1.4.2. My suggestion can be to support the last version of Flink1.17/1.16/1.15 or Spark 3.2/3.3/3.4

### How should we improve?

- upgrade iceberg pom to 1.4.2
- remove spark 3.1 support

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by XBaith**
## Why are the changes needed?

Close #2410.

## Brief change log

- Bump iceberg to 1.4.3
- Support config storage class for S3FileIO
- Dropped support for Spark 3.1

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: .rat-excludes [hunk 1]
```diff
@@ -4,6 +4,7 @@
 **/*.md
 **/*.iml
 **/target/**
+**/dist/**
```
Comment (zhoujinsong) [inline .rat-excludes:7]: Directory `ams/dist` contains some files that need to be checked, we may not ignore them all.

**Review by zhoujinsong — COMMENTED** [review]
@XBaith Thanks a lot for your contribution.

We may want to start a discussion about abandoning support for the Spark 3.1 version of Mixed format before we commit this.

File: .rat-excludes (hunk 1 above)
Comment (XBaith) [reply .rat-excludes]: make sense, some files will be added after building project. I will only exclude the font files

**Review by zhoujinsong — APPROVED** [review]
I tested it in my local envrionment followed the quick start process and it works fine.

Thanks a lot for your great work!

**Pull request merged**